jgrapht-core/src/main/java/org/jgrapht/Graph.java

```diff
@@ -255,6 +249,7 @@ public interface Graph<V, E>
      *
      * @return {@code true} if this graph did not already contain the specified vertex.
      *
+     * @throws IllegalArgumentException if there is a property that disallows adding the specified vertex (optional operation)
```

Comment:
Is there actually a case where this happens?  I think the IllegalArgumentException only gets thrown by the no-args method, right?

Reply:
https://github.com/jgrapht/jgrapht/blob/6a6f167f4c53a9bb2d4f00dcf11e6555435191c8/jgrapht-core/src/main/java/org/jgrapht/graph/AsGraphUnion.java#L199-L203

https://github.com/jgrapht/jgrapht/blob/6a6f167f4c53a9bb2d4f00dcf11e6555435191c8/jgrapht-core/src/main/java/org/jgrapht/graph/AsUnmodifiableGraph.java#L90-L94

https://github.com/jgrapht/jgrapht/blob/6a6f167f4c53a9bb2d4f00dcf11e6555435191c8/jgrapht-core/src/main/java/org/jgrapht/graph/AsSubgraph.java#L289-L299

There are at least three implementations of the `Graph` interface that has this behavior, so I think it is still worth documeting.

Reply:
Ah I see, you're right!  Should we delete the "(optional operation)"?  I don't think it applies here.

Reply:
I don't have a strong opinion on this and think removing it wouldn't hurt clarity. eb7f824 removes it.